newIDE/app/src/LoginProvider/LocalLoginProvider.js

```diff
@@ -47,11 +47,24 @@ class LocalLoginProvider implements LoginProvider, FirebaseBasedLoginProvider {
 
   async loginOrSignupWithProvider({
     provider,
+    signal,
   }: {|
     provider: IdentityProvider,
     loginOptions?: ?LoginOptions,
+    signal?: AbortSignal,
   |}) {
+    if (signal && signal.aborted) {
+      return Promise.reject(
+        new Error('Login or Signup with provider already aborted.')
+      );
+    }
     const promise = new Promise((resolve, reject) => {
+      // Listen for abort event on signal
+      if (signal) {
+        signal.addEventListener('abort', () => {
+          reject(new Error('Login or Signup with provider aborted.'));
```

Comment:
You might reject here, but the onTokenReceived/onConnectionEstablished/onError could still be called. So I think you need to check for `signal && signal.aborted` instead in each callback and just "do nothing".

The best we can do with callbacks (or Promise) is to just ignore things?